### ISSUE TITLE
Philosophy of the master reference

### ISSUE THREAD

**Issue by mattlub** [foundersandcoders/master-reference#591]
#### Do we have a clearly defined plan/vision/philosophy for the master reference?

If not, do we not need one? I know we have the "vision.md" file but as far as I'm aware it is outdated.

**Why do we need one?**
- If we don't have a clearer vision we can't define what it even means to improve the curriculum
- This greatly affects curriculum planning which is being discussed heavily right now
- what is a valid reason to raise an issue on a workshop/morning challenge/week of the MR? 

I think different people have ideas of different philosophies, including:

- Accessibility: 
The master reference should be a curriculum which anyone should be able to follow (without mentors), so mentor dependent activities (code-along, presentations) should not exist, and workshops which require lots of mentor input should be changed/improved. 

- Making the 8-week F&C course as good as possible for the students on it:
code-alongs and other mentor-dependent activities are encouraged, as they are beneficial to the students. Workshops should be pitched at the right level

- Modularity:
Each week (or other section) could be a standalone mini-course, so workshops are mainly technology independent

**Basically, do we need to decide on a vision/philosophy/plan/aim for the MR, and if so, what should it be?**

**Comment by jsms90** [comment]
This relates directly to what we want the "[course learning outcomes](https://github.com/foundersandcoders/master-reference/issues/48)" to be.

@mattlub can we change the title to "philosohpy of curriculum design"?

The "master reference" could refer to lots of things e.g.
+ the aims listed in [the vision.md](https://github.com/foundersandcoders/master-reference/blob/master/vision.md)
+ the implementation
+ meta level processes e.g. labels (how we collaborate via github) or stand-ups (how we collaborate in person)

As for:
> The master reference should be a curriculum which anyone should be able to follow 

this relates to the idea of having an "online version of the curriculum" #65. There's nothing wrong with the occasional mentor-led activity, so long as it is the best way for students to learn that material.

But for me, the core reason behind wanting workshops to stand alone is not about "accessibility" as much as it is about a belief in:
+ peer-led learning (hence using the word "mentoring" vs "teaching" - #571 )
+ "[active learning](https://www.cambridge-community.org.uk/professional-development/gswal/index.html)" / "[the inductive approach](https://www.slideshare.net/itsmemajeanrm/inductive-method)" / "[hands-on learning](https://www.readorium.com/2015/04/hands-on-learning-and-guided-discovery/)" / "[guided discovery](http://onlignment.com/strategies-for-learning-guided-discovery/)" 

because:
> Actively engaging students in the classroom will help
them think more deeply about the course content,
bring additional energy to the classroom, and help
identify the extent to which they may be struggling
with the material

(from ["Classroom activities for active learning"](http://cfe.unc.edu/files/2014/08/FYC2.pdf))

**Comment by mattlub** [comment]
@jsms90 I changed from "Master Reference" to "curriculum".

Thanks for the answer, I agree, I just think we need clearer things to aim for. It could be as simple as a bullet point list of beliefs that we agree on, but I believe we need it because then you can have something to refer to when creating/updating a resource/section of the curriculum.

i.e. "this workshop needs improvement because it goes against \<some aim of the curriculum\>"

**Comment by astroash** [comment]
I disapprove of this.

**Comment by jsms90** [comment]
Just added this to [the issue on mentor expectations](https://github.com/foundersandcoders/master-reference/issues/646#issuecomment-333312668):
> In general, I think that documentation should be an essential part of a mentor's responsibilities. We can only improve FAC's curriculum together if mentors ensure that whatever material they create/deliver can be be picked up and delivered to the same standard by any future mentor.

That's why I take the documention our curriculum in the `master-reference` so seriously. In particular, when it comes to any mentor-led content #591 (e.g. tips in a `mentor-notes` folder #456, a screencast for code-alongs like @oliverjam's TDD one, and a video of any presentations like @stevehopkinson's database talk)
